Classes/common/HTTP/CDTURLSessionTask.h

```diff
@@ -0,0 +1,24 @@
+//
+//  CDTURLSessionTask.h
+//  
+//
+//  Created by Rhys Short on 20/08/2015.
+//
+//
+#import <Foundation/Foundation.h>
+#import "CDTMacros.h"
+
+@interface CDTURLSessionTask : NSObject
+
+@property (readonly) NSURLSessionTaskState state;
+@property (nonnull,readonly) NSURLSessionTask *task;
+
+- (nullable instancetype) init UNAVAILABLE_ATTRIBUTE;
+- (nullable instancetype) initWithTask:(nonnull NSURLSessionTask *) task NS_DESIGNATED_INITIALIZER;
+
+
+- (void) resume;
+- (NSURLSessionTaskState) state;
+- (void) cancel;
```

Comment:
In these cases they are just proxies through to the `NSURLSessionTask` method of the same name. I think the documentation should come later when the class holds the information needed to enable retries, and these methods will use that information to correctly determine, state etc when called.